app/components/trigger-custom-build.js

```diff
@@ -0,0 +1,89 @@
+import Ember from 'ember';
+import { task } from 'ember-concurrency';
+import YAML from 'npm:yamljs';
+
+const { service } = Ember.inject;
+
+export default Ember.Component.extend({
+  ajax: service(),
+  flashes: service(),
+  router: service(),
+
+  classNames: ['trigger-build-modal'],
+  triggerBuildBranch: '',
+  triggerBuildMessage: '',
+  triggerBuildConfig: '',
+  triggering: false,
+
+  branches: Ember.computed.filterBy('repo.branches', 'exists_on_github', true),
+
+  didReceiveAttrs() {
+    this._super(...arguments);
+    this.set('triggerBuildBranch', this.get('repo.defaultBranch.name'));
+  },
+
+  sendTriggerRequest: task(function* () {
+    let config = YAML.parse(this.get('triggerBuildConfig'));
+
+    let runInterval = 0;
+    if (!Ember.testing) {
+      runInterval = 3000;
+    }
+
+    let body = {
+      request: {
+        branch: this.get('triggerBuildBranch'),
+        config: config
+      }
+    };
+
+    if (! Ember.isEmpty(this.get('triggerBuildMessage'))) {
+      body.request.message = this.get('triggerBuildMessage');
+    }
+
+    try {
+      yield this.get('ajax').postV3(`/repo/${this.get('repo.id')}/requests`, body)
```

Comment:
hmmmm maybe you need a `return` inside the `Ember.run.later`? Then it should get added to the promise chain and be considered part of the task.

But now that I think about it, maybe there should be more `yield`s here, like in [this example](http://ember-concurrency.com/#/docs/examples/autocomplete)…??? I think there could be some flattening out of this getting-to-pyramid-esque-ness